### ISSUE TITLE
feature: strong modularity boundaries between KubeFlex and ks-0.20

### ISSUE THREAD

**Issue by MikeSpreitzer** [kubestellar/kubestellar#1533]
### Feature Description

Almost all of ks-0.20 does not depend on KubeFlex. Almost all of KubeFlex makes no mention of ks-0.20. ks-0.20 should be usable in a simple deployment scenario that does not involve KubeFlex, using one cluster as WDS and IS and mailbox namespace holder. The only connections between KubeFlex and ks-0.20 are conveniences that could be moved out to a third place, leaving KubeFlex and ks-0.20 orthogonal to each other.

### Proposed Solution

details TBD

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by ezrasilvera** [comment]
Agree. The plan is to support a flow in which we separate the creation of spaces and the deployment into them (e.g., of the transport and the 0.20 controller(s) ) .

**Comment by pdettori** [comment]
Agreed on the orthogonality. I think though we may still allow to option to use the KF's post-create-hook to install the ks 0.20 controller. User may use that when creating the wds or may not use the -p option and then run helm to install the controller.

**Comment by MikeSpreitzer** [comment]
@pdettori : yes, and the PostCreateHook that installs the ks-0.20 controller does not need to be in the kubeflex repo, right? This is part of what I said about the connections between the two being something that can be a third module that depends on KubeFlex and ks-0.20.

**Comment by pdettori** [comment]
@MikeSpreitzer 
> the PostCreateHook that installs the ks-0.20 controller does not need to be in the kubeflex repo, right?

It does not, but we may need to find a 3rd party repo where to host the yaml for the hook, currently is in https://github.com/kubestellar/kubeflex/blob/main/config/samples/postcreate-hooks/kslight.yaml

**Comment by nirrozenbaum** [comment]
totally agree on the orthogonality and also with Ezra that we should separate the creation of spaces and the deployment into them (as it was in SF).

**Comment by ezrasilvera** [comment]
@MikeSpreitzer This is phase1 in my opinion. You can say high priority phase1.  People can start working on phase1 items while this is being implemented.

**Comment by andreyod** [comment]
To make the Kubestellar operator independent from Kubeflex we need a way to provide the access(kubeconfig) to WDS and IMBS. The initial proposal is to use secrets. Kubeflex is already creates those secrets, and for other scenarios, the user(script) will be responsible to create them in the same way(name/namespace).

**Comment by pdettori** [comment]
@andreyod I am ok in using secrets vs. discovery based on CPs, as long as we can still support the deployment model based on post-create-hooks. For the WDS, should be pretty simple, as we pass the name of the WDS, for the IMBS we could do the same, but if in the future we want to use sharding (multiple IMBS shards) we may have to explicitly provide the names/secrets for all of them rather than the current approach that relies on discovery based on labels.

**Comment by andreyod** [comment]
Good point. In the future, for multiple IMBS shards, we can change the parameter(s) to be a list.
@ezrasilvera @MikeSpreitzer WDYT?

**Comment by MikeSpreitzer** [comment]
I am not clear on what the proposal is. I am pretty confident that there will be a path forward. I think that anything fixed on a captured file is not going to be acceptable long term, because the credentials captured in a file typically last for an hour or a year. We will need a different technology to make a system that lasts for more than a year. I am hoping to hear good things soon from colleagues working with SPIFFE/SPIRE.

**Comment by clubanderson** [comment]
closing - we are at v 0.28 at this time
